src/core/server/rendering/rendering_service.tsx

```diff
@@ -95,9 +95,11 @@ export class RenderingService {
           defaults: uiSettings.getRegistered(),
           user: includeUserSettings ? await uiSettings.getUserProvided() : {},
         };
-        const darkMode = settings.user?.['theme:darkMode']?.userValue
-          ? Boolean(settings.user['theme:darkMode'].userValue)
-          : false;
+        // Cannot use `uiSettings.get()` since a user might not be authenticated
+        const darkMode =
+          (settings.user?.['theme:darkMode']?.userValue ??
+            uiSettings.getOverrideOrDefault('theme:darkMode')) ||
+          false;
```

Comment:
From my read of the tests, I don't think we actually test this ultimate fallback of `false`. Do you think we should?

Reply:
I didn't think it was helpful but I added one anyway.